src/Core/Checkout/Promotion/Cart/Discount/Filter/AdvancedPackageRules.php

```diff
@@ -80,4 +66,52 @@ private function isRulesFilterValid(LineItem $item, PriceDefinitionInterface $pr
 
         return false;
     }
+
+    /**
+     * @return array{0: LineItemQuantity[], 1: LineItem[]}
```

Comment:
```suggestion
     * @return array{array<string, LineItemQuantity>, array<string, LineItem>}
```

this annotation is more precise, the `[]` annotation could have several meanings